### DISCUSSION TITLE
Clarification of the behaviour of `refetchOnMountOrArgChange`

### DISCUSSION THREAD

**Discussion by dwjohnston** [reduxjs/redux-toolkit#3654] [General]
The `refetchOnMountOrArgChange` option is helpful for me, where when a component remounts I want the data the refetched regardless of what's already in the cache. 

However, I was worried that if I had something like: 

```typescript
  const result = useGetPostQuery(
    { id: id }, //<-- This line here worries me
    {
      pollingInterval: globalPolling ? 3000 : 0,
      refetchOnMountOrArgChange: true
    }
  );
```

Then you would get a 'refetch every render' problem - because a new object is declared every time this component/hook renders. 

However, this doesn't seem to be a problem. 

I've a repro of this here, it's a modification of the Kitchen Sink example. 

https://codesandbox.io/s/lucid-haze-qqntsp?file=/src/features/posts/PostDetail.tsx:1779-1963

1. Go to posts section, click between the 'A sample post' and 'A post about rtk-query' posts. Note that they refetch, which is the 'on mount refetch behaviour we want'
2. Click the rerender button. Note there is no refetch, which is also what we want. 

I'm wondering how is this behaviour achieved? Is RTKQ stringifying the object to check for deep equality?

**Comment by dwjohnston** [comment]
It does look like that's the case. Here's the args serialisation function: 

https://github.com/reduxjs/redux-toolkit/blob/e45425128d3c3168c7daa71e5f38f5151234cb8d/packages/toolkit/src/query/defaultSerializeQueryArgs.ts#L9-L37

Is this a potential problematic performance issue, if we had large args objects that happened to be going through as args, and they'd be doing that sort and stringify every render? 

(But FWIF, at this point I'm not griping at all - one of my biggest bug bears with React is how the declaration of new objects will trip shallow equality checking and cause all sorts of issues for me, so that this doesn't, is quite a relief for me).

**Reply by phryneas** [reply]
If you rerender an unreasonable amount of times and have a very big object, it might be a problem, but usually it shouldn't be - and in that case you could still use useMemo for referential stability on your side.
